### PR TITLE
3d test new

### PR DESCRIPTION
**Pull request by shyeavr**
Add Yoad test, generate points by polygon function , retype env function

### CODE REVIEW

File: common/utils/data_generator/data_utils.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import json
+import random
+
+from shapely.geometry import Point, Polygon
+
```
Comment (dannysigalovich) [inline common/utils/data_generator/data_utils.py:5]: # Define constants
PRODUCT_TYPES = ["DSM", "DTM", "MIXED"]

File: common/utils/data_generator/data_utils.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+poly = [
+    (37.75850848099701, -122.50833008408812),
+    (37.75911919711413, -122.49648544907835),
+    (37.751620611284935, -122.4937388670471),
+    (37.74863453749236, -122.50742886185911),
+]
+
+
+def polygon_random_points(polygon: list, num_points: int) -> list:
```
Comment (dannysigalovich) [inline common/utils/data_generator/data_utils.py:14]:
```
def polygon_random_points(polygon: List[Tuple[float, float]], num_points: int) -> List[Tuple[float, float]]:

double check me
```

File: common/utils/data_generator/data_utils.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        random_point = Point(
+            [random.uniform(min_x, max_x), random.uniform(min_y, max_y)]
+        )
+        if random_point.within(polygon):
+            point_val = (float(random_point.x), float(random_point.y))
+            points.append(point_val)
+    return points
+
+
+def generate_points_request(
```
Comment (dannysigalovich) [inline common/utils/data_generator/data_utils.py:35]: points_amount: int, polygon: List[Tuple[float, float]], exclude_fields: bool, product_type: str = "MIXED"

Make it harder - double check me

File: common/utils/data_generator/data_utils.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    This function will generate request body for user
+    :param points_amount: point amount to be generated
+    :param poly: Polygon object of the desired points zone
+    :param payload_flag: indicate if the excludeFields will be empty or not
+    :param product_type: [ DSM, DTM, MIXED ]
+    :return:
+    request body
+    """
+    points_data = polygon_random_points(polygon=poly, num_points=points_amount)
+    points_list = []
```
Comment (dannysigalovich) [inline common/utils/data_generator/data_utils.py:48]: points_list = [{"longitude": lon, "latitude": lat} for lon, lat in points_data]

Comment (dannysigalovich) [reply common/utils/data_generator/data_utils.py:48]: it replace the for

File: common/utils/data_generator/data_utils.py (hunk 3 above)
Comment (dannysigalovich) [reply common/utils/data_generator/data_utils.py:35]:
```
def generate_points_request(
    points_amount: int, polygon: List[Tuple[float, float]], exclude_fields: bool, product_type: str = "MIXED"
) 
```

File: common/utils/data_generator/data_utils.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    for points_cor in points_data:
+        points = {"longitude": points_cor[0], "latitude": points_cor[1]}
+        points_list.append(points)
+    if payload_flag:
```
Comment (dannysigalovich) [inline common/utils/data_generator/data_utils.py:52]:
```
if exclude_fields:
        request_body = {
            "positions": points_list,
            "productType": product_type,
            "excludeFields": []
        }
    else:
        request_body = {
            "positions": points_list,
            "productType": product_type,
            "excludeFields": ["productType", "updateDate", "resolutionMeter"]
        }

    return json.dumps(request_body)
```

File: tests/elevation_service/one_point_per_user.py [hunk 6]
```diff
@@ -1,131 +1,164 @@
...
+    global total_requests, test_results, counters
+    average_response_time = environment.runner.stats.total.avg_response_time
+    test_results.append(
+        {"users": environment.users_count, "avg_response_time": average_response_time}
+    )
+    create_custom_graph(
+        graph_name="Users_vs_AvgResponseTime",
+        graph_path=reports_path,
+        test_results=test_results,
+        graph_title=None,
```
Comment (dannysigalovich) [inline tests/elevation_service/one_point_per_user.py:149]: The graph_title parameter of create_custom_graph is set to None. If there is a specifisc title for the graph make it clarity!

File: tests/elevation_service/one_point_per_user.py [hunk 7]
```diff
@@ -1,131 +1,164 @@
...
-    wait_time = constant(ElevationConfig.wait_time)
+# @events.test_start.add_listener
+# def reset_counters(**kwargs):
+#     global counters, total_requests
+#     counters = counters
+#     total_requests = 0
+
 
 @events.test_stop.add_listener
 def on_locust_stop(environment, **kwargs):
```
Comment (dannysigalovich) [inline tests/elevation_service/one_point_per_user.py:139]: the code lacks comments and docstrings

File: tests/elevation_service/one_point_per_user.py [hunk 8]
```diff
@@ -1,131 +1,164 @@
...
+    )
+    create_custom_graph(
+        graph_name="Users_vs_AvgResponseTime",
+        graph_path=reports_path,
+        test_results=test_results,
+        graph_title=None,
+    )
+    percent_value_by_range = {}
+    for index, (key, value) in enumerate(counters.items()):
+        percent_range = (value / total_requests) * 100
```
Comment (dannysigalovich) [inline tests/elevation_service/one_point_per_user.py:153]: Magic number? what is 100

File: tests/elevation_service/one_point_per_user.py [hunk 9]
```diff
@@ -1,131 +1,164 @@
...
-def on_request(response_time, **kwargs):
-    start_time = datetime.datetime.fromtimestamp(kwargs['start_time'])
-    response_time_data.append(response_time)
-    start_time_data.append(start_time)
+
+counters = initiate_counters_by_ranges(config_ranges=percent_ranges)
 
 
 @events.request.add_listener
 def response_time_listener(response_time, **kwargs):
```
Comment (dannysigalovich) [inline tests/elevation_service/one_point_per_user.py:87]: General comment : *kwargs is not good practice, better us meaningful names for function param :)

File: tests/elevation_service/one_point_per_user.py [hunk 10]
```diff
@@ -1,131 +1,164 @@
...
+#     response_time_data.append(response_time)
+#     start_time_data.append(start_time)
+#     total_requests += 1
+
 
 @events.test_start.add_listener
-def reset_counters(**kwargs):
-    global counters, total_requests, run_number, start_time_data, response_time_data
-    counters = counters
+def reset_data_counters(**kwargs):
```
Comment (dannysigalovich) [inline tests/elevation_service/one_point_per_user.py:124]: General comment : *kwargs is not good practice, better us meaningful names for function param :)

File: tests/elevation_service/elevation-integ.py [hunk 11]
```diff
@@ -26,13 +26,18 @@ def __init__(self, *args, **kwargs):
     def index(self):
```
Comment (dannysigalovich) [inline tests/elevation_service/elevation-integ.py:26]:
```
def index(self):
    payload_type = self.request_body["request_type"]
    payload = self.request_body["body"]
    headers = self.request_body["header"]
    verify = False 

    if payload_type == "json":
        self.client.post("/", json=payload, headers=headers, verify=verify)
    elif payload_type == "bin":
        self.client.post("/", data=payload, headers=headers, verify=verify)
```

File: common/utils/data_generator/data_utils.py (hunk 1 above)
Comment (dannysigalovich) [reply common/utils/data_generator/data_utils.py:5]: not sure

**Review by dannysigalovich — CHANGES_REQUESTED** [review]
@shyeavr please fix the code :) double check me !

**Review by dannysigalovich — APPROVED** [review]

**Comment by shyeavr** [comment]
change all comments

**Pull request closed without merging**